pkg/controllers/nodetemplate/controller.go

```
@@ -73,21 +83,32 @@ func (c *Controller) Builder(ctx context.Context, m manager.Manager) corecontrol
 		NewControllerManagedBy(m).
 		For(&v1alpha1.AWSNodeTemplate{}).
 		WithEventFilter(predicate.GenerationChangedPredicate{}).
-		WithOptions(controller.Options{MaxConcurrentReconciles: 10}))
+		WithOptions(controller.Options{
+			RateLimiter: workqueue.NewMaxOfRateLimiter(
+				workqueue.NewItemExponentialFailureRateLimiter(100*time.Millisecond, 1*time.Minute),
+				// 10 qps, 100 bucket size
+				&workqueue.BucketRateLimiter{Limiter: rate.NewLimiter(rate.Limit(10), 100)},
+			),
+			MaxConcurrentReconciles: 10,
+		}))
 }
 
 func (c *Controller) resolveSubnets(ctx context.Context, nodeTemplate *v1alpha1.AWSNodeTemplate) error {
 	subnetList, err := c.subnetProvider.List(ctx, nodeTemplate)
 	if err != nil {
 		return err
 	}
+	if len(subnetList) == 0 {
+		nodeTemplate.Status.Subnets = nil
+		logging.FromContext(ctx).Errorf("no subnet exist given constraints")
```

We should not just log an error here, I think we want to return an error